src/pages/signin/SignInPage.js

```diff
@@ -44,16 +46,32 @@ const propTypes = {
         // Error to display when there is a session error returned
         authToken: PropTypes.string,
     }),
+
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            validateCode: PropTypes.string,
+        }),
+    }),
 };
 
 const defaultProps = {
     account: {},
     session: {},
     credentials: {},
+    route: {
+        params: {},
+    },
 };
 
 class SignInPage extends Component {
     componentDidMount() {
+        // If we've redirected to the signin page with a validation code, validate the login
```

Comment:
Can the login be validated inside the action in the first place before redirecting? It seems strange to have an action redirect to a page, just to have the new page call another method in the same action.

Reply:
Hmm so we navigate to SingInPage.js from [PublicScreens.js](src/libs/Navigation/AppNavigator/PublicScreens.js)... I suppose we don't need to redirect here to do that and instead could it in PublicScreens.js. 

To be honest, I'm not sure what the best flow is here - ideally, we'd run this immediately since we want it to happen regardless of whether or not the user is currently logged in, but we don't have any other logic like that at the moment (as far I know of). 

Definitely open to exploring other implementations

Reply:
Ah, OK. Sorry, I misunderstood the redirect from `User.validateLogin()`.

I think that `PublicScreens.js` should have a route handling specifically the validate login functionality, then once that's done, let it redirect to SignIn. Does that make sense? I think that's inline with the other comments about using explicit routes.